src/mito2/src/memtable/merge_tree/shard_builder.rs

```diff
@@ -107,10 +108,21 @@ impl ShardBuilder {
 
     /// Scans the shard builder.
     pub fn read(&self, pk_weights_buffer: &mut Vec<u16>) -> Result<ShardBuilderReader> {
-        let dict_reader = self.dict_builder.read();
-        dict_reader.pk_weights_to_sort_data(pk_weights_buffer);
-        let data_reader = self.data_buffer.read(Some(pk_weights_buffer))?;
+        let dict_reader = {
+            let _timer = MERGE_TREE_READ_STAGE_ELAPSED
+                .with_label_values(&["shard_builder_read_pk"])
+                .start_timer();
+            self.dict_builder.read()
+        };
 
+        {
+            let _timer = MERGE_TREE_READ_STAGE_ELAPSED
+                .with_label_values(&["sort_pk"])
+                .start_timer();
+            dict_reader.pk_weights_to_sort_data(pk_weights_buffer);
+        }
+
+        let data_reader = self.data_buffer.read(Some(pk_weights_buffer))?;
```

Comment:
Would this be costly?

Reply:
> Would this be costly?

Cost time of `self.data_buffer.read` will be recorded here: https://github.com/GreptimeTeam/greptimedb/blob/8a1736bc7931ec9ffaf44962aff46713c5f0bc91/src/mito2/src/memtable/merge_tree/data.rs#L259-L263